apps/site/app/[locale]/download/archive/[version]/page.tsx

```diff
@@ -0,0 +1,79 @@
+import { notFound } from 'next/navigation';
+import type { FC } from 'react';
+
+import { ENABLE_STATIC_EXPORT } from '#site/next.constants.mjs';
+import { ARCHIVE_DYNAMIC_ROUTES } from '#site/next.dynamic.constants.mjs';
+import * as basePage from '#site/next.dynamic.page.mjs';
+import { defaultLocale } from '#site/next.locales.mjs';
+
+type DynamicStaticPaths = { version: string; locale: string };
+type DynamicParams = { params: Promise<DynamicStaticPaths> };
+
+// This is the default Viewport Metadata
+// @see https://nextjs.org/docs/app/api-reference/functions/generate-viewport#generateviewport-function
+export const generateViewport = basePage.generateViewport;
+
+// This generates each page's HTML Metadata
+// @see https://nextjs.org/docs/app/api-reference/functions/generate-metadata
+export const generateMetadata = basePage.generateMetadata;
+
+// Generates all possible static paths based on the locales and environment configuration
+// - Returns an empty array if static export is disabled (`ENABLE_STATIC_EXPORT` is false)
+// - If `ENABLE_STATIC_EXPORT_LOCALE` is true, generates paths for all available locales
+// - Otherwise, generates paths only for the default locale
+// @see https://nextjs.org/docs/app/api-reference/functions/generate-static-params
+export const generateStaticParams = async () => {
+  // Return an empty array if static export is disabled
+  if (!ENABLE_STATIC_EXPORT) {
+    return [];
+  }
+
+  return ARCHIVE_DYNAMIC_ROUTES.map(pathname => ({
+    locale: defaultLocale.code,
+    version: pathname,
+  }));
+};
+
+// This method parses the current pathname and does any sort of modifications needed on the route
+// then it proceeds to retrieve the Markdown file and parse the MDX Content into a React Component
+// finally it returns (if the locale and route are valid) the React Component with the relevant context
+// and attached context providers for rendering the current page
+const getPage: FC<DynamicParams> = async props => {
+  const { version, locale: routeLocale } = await props.params;
+
+  // Gets the current full pathname for a given path
+  const [locale, pathname] = basePage.getLocaleAndPath(version, routeLocale);
+
+  // Verifies if the current route is a dynamic route
+  const isDynamicRoute = ARCHIVE_DYNAMIC_ROUTES.some(r => r.includes(pathname));
+
+  // Gets the Markdown content and context for Download Archive pages
+  const [content, context] = await basePage.getMarkdownContext({
+    locale: locale,
+    pathname: 'download/archive',
+  });
+
+  // If this isn't a valid dynamic route for archive version or there's no markdown
+  //  file for this, then we fail as not found as there's nothing we can do.
+  if (isDynamicRoute && context.filename) {
+    return basePage.renderPage({
+      content: content,
+      layout: context.frontmatter.layout ?? 'download-archive',
```

Comment:
```suggestion
      layout: context.frontmatter.layout,
```

I believe here a layout is always coming from frontmatter, right?